modules/prebidServerBidAdapter.js

```diff
@@ -272,126 +261,289 @@ export function PrebidServer() {
+          bidObject.height = bid.h;
+          if (bid.dealid) { bidObject.dealId = bid.dealid; }
+          bidObject.requestId = bid.id;
+          bidObject.creative_id = bid.crid;
+          bidObject.creativeId = bid.crid;
+
+          // TODO: Remove when prebid-server returns ttl, currency and netRevenue
+          bidObject.ttl = (bid.ttl) ? bid.ttl : DEFAULT_S2S_TTL;
+          bidObject.currency = (bid.currency) ? bid.currency : DEFAULT_S2S_CURRENCY;
+          bidObject.netRevenue = (bid.netRevenue) ? bid.netRevenue : DEFAULT_S2S_NETREVENUE;
+
+          bids.push({ adUnit: bid.impid, bid: bidObject });
         });
+      });
+    }
 
-        if (result.bids) {
-          result.bids.forEach(bidObj => {
-            let bidRequest = utils.getBidRequest(bidObj.bid_id, bidRequests);
-            let cpm = bidObj.price;
-            let status;
-            if (cpm !== 0) {
-              status = STATUS.GOOD;
-            } else {
-              status = STATUS.NO_BID;
-            }
+    return bids;
+  }
+};
 
-            let bidObject = bidfactory.createBid(status, bidRequest);
-            bidObject.source = TYPE;
-            bidObject.creative_id = bidObj.creative_id;
-            bidObject.bidderCode = bidObj.bidder;
-            bidObject.cpm = cpm;
-            if (bidObj.cache_id) {
-              bidObject.cache_id = bidObj.cache_id;
-            }
-            if (bidObj.cache_url) {
-              bidObject.cache_url = bidObj.cache_url;
-            }
-            // From ORTB see section 4.2.3: adm Optional means of conveying ad markup in case the bid wins; supersedes the win notice if markup is included in both.
-            if (bidObj.media_type === VIDEO) {
-              bidObject.mediaType = VIDEO;
-              if (bidObj.adm) {
-                bidObject.vastXml = bidObj.adm;
-              }
-              if (bidObj.nurl) {
-                bidObject.vastUrl = bidObj.nurl;
-              }
-            } else {
-              if (bidObj.adm && bidObj.nurl) {
-                bidObject.ad = bidObj.adm;
-                bidObject.ad += utils.createTrackPixelHtml(decodeURIComponent(bidObj.nurl));
-              } else if (bidObj.adm) {
-                bidObject.ad = bidObj.adm;
-              } else if (bidObj.nurl) {
-                bidObject.adUrl = bidObj.nurl
-              }
-            }
+/*
+ * Returns the required protocol adapter to communicate with the configured
+ * endpoint. The adapter is an object containing `buildRequest` and
+ * `interpretResponse` functions.
+ *
+ * Usage:
+ * // build JSON payload to send to server
+ * const request = protocol().buildRequest(s2sBidRequest, adUnits);
+ *
+ * // turn server response into bid object array
+ * const bids = protocol().interpretResponse(response, bidRequests, requestedBidders);
+ */
+const protocolAdapter = () => {
+  const OPEN_RTB_PATH = 'openrtb2/auction';
+  const isOpenRtb = _s2sConfig.endpoint.includes(OPEN_RTB_PATH);
```

Comment:
One of the legacy test pages I found originally had an endpoint like `https://prebid.adnxs.com/pbs/v1/auction?url_override=...`, so checking for the end of the string might not work. Good point about the polyfill, updated to use old school `indexOf` rather than the string version of `includes`